backend/cmm_helpers.ml

```diff
@@ -2070,7 +2070,43 @@ let generic_apply mut clos args (pos, mode) dbg =
     let cargs =
       (Cconst_symbol (apply_function_sym arity mode, dbg) :: args) @ [clos]
     in
-    Cop (Capply (typ_val, pos), cargs, dbg)
+    let call_caml_apply = Cop (Capply (typ_val, pos), cargs, dbg)  in
+    if !Flambda_backend_flags.caml_apply_inline_fast_path then
+      (*    (if (= clos.arity N)
+       *      (app clos.direct a1 ... aN clos)
+       *      call_caml_applyN)
+      *)
+      (* CR gyorsh: Should we use
+         [Asttypes.Mutable] and [Rc_normal]
+         instead of [mut] and [pos] in the code below?
+         In [apply_function_body], the former is used.
+         In the single argument case above, the latter is used.
+      *)
+      (* CR gyorsh: [args] are used twice below, do we need to "bind" them all? *)
```

Comment:
I think that you indeed need to "bind" them all, else you can end up duplicating some expressions, which is a problem, particularly considering that it is preferable to produce nested expressions in cmm (due to the way optimisations are applied).